In [1]:
import pandas as pd
import numpy as np
import datetime

In [2]:
intakes = pd.read_csv('aac_intakes.csv')
intakes_outcomes = pd.read_csv('aac_intakes_outcomes.csv')
outcomes = pd.read_csv('aac_outcomes.csv')

In [3]:
intakes.head(3)

age_upon_intake animal_id animal_type                     breed  \
0         8 years   A706918         Dog  English Springer Spaniel   
1       11 months   A724273         Dog               Basenji Mix   
2         4 weeks   A665644         Cat    Domestic Shorthair Mix   

         color                 datetime                datetime2  \
0  White/Liver  2015-07-05T12:59:00.000  2015-07-05T12:59:00.000   
1  Sable/White  2016-04-14T18:43:00.000  2016-04-14T18:43:00.000   
2       Calico  2013-10-21T07:59:00.000  2013-10-21T07:59:00.000   

                       found_location intake_condition intake_type     name  \
0    9409 Bluegrass Dr in Austin (TX)           Normal       Stray    Belle   
1  2818 Palomino Trail in Austin (TX)           Normal       Stray  Runster   
2                         Austin (TX)             Sick       Stray      NaN   

  sex_upon_intake  
0   Spayed Female  
1     Intact Male  
2   Intact Female

In [4]:
intakes_outcomes.head(3)

age_upon_outcome animal_id_outcome        date_of_birth outcome_subtype  \
0         10 years           A006100  2007-07-09 00:00:00             NaN   
1          7 years           A006100  2007-07-09 00:00:00             NaN   
2          6 years           A006100  2007-07-09 00:00:00             NaN   

      outcome_type sex_upon_outcome  age_upon_outcome_(days)  \
0  Return to Owner    Neutered Male                     3650   
1  Return to Owner    Neutered Male                     2555   
2  Return to Owner    Neutered Male                     2190   

   age_upon_outcome_(years) age_upon_outcome_age_group     outcome_datetime  \
0                      10.0                (7.5, 10.0]  2017-12-07 14:07:00   
1                       7.0                 (5.0, 7.5]  2014-12-20 16:35:00   
2                       6.0                 (5.0, 7.5]  2014-03-08 17:10:00   

   ...  age_upon_intake_age_group      intake_datetime intake_month  \
0  ...                (7.5, 10.0]  2017-12-07 00:00:00           12   
1  ...                 (5.0, 7.5]  2014-12-19 10:21:00           12   
2  ...                 (5.0, 7.5]  2014-03-07 14:26:00            3   

  intake_year  intake_monthyear  intake_weekday  intake_hour  intake_number  \
0        2017           2017-12        Thursday           14            1.0   
1        2014           2014-12          Friday           10            2.0   
2        2014           2014-03          Friday           14            3.0   

             time_in_shelter time_in_shelter_days  
0  0 days 14:07:00.000000000             0.588194  
1  1 days 06:14:00.000000000             1.259722  
2  1 days 02:44:00.000000000             1.113889  

[3 rows x 41 columns]

In [5]:
outcomes.head(3)

age_upon_outcome animal_id animal_type                   breed  \
0          2 weeks   A684346         Cat  Domestic Shorthair Mix   
1           1 year   A666430         Dog              Beagle Mix   
2           1 year   A675708         Dog                Pit Bull   

          color        date_of_birth             datetime  \
0  Orange Tabby  2014-07-07T00:00:00  2014-07-22T16:04:00   
1   White/Brown  2012-11-06T00:00:00  2013-11-07T11:47:00   
2    Blue/White  2013-03-31T00:00:00  2014-06-03T14:20:00   

             monthyear     name outcome_subtype outcome_type sex_upon_outcome  
0  2014-07-22T16:04:00      NaN         Partner     Transfer      Intact Male  
1  2013-11-07T11:47:00     Lucy         Partner     Transfer    Spayed Female  
2  2014-06-03T14:20:00  *Johnny             NaN     Adoption    Neutered Male

In [6]:
intakes.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 80187 entries, 0 to 80186
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   age_upon_intake   80187 non-null  object
 1   animal_id         80187 non-null  object
 2   animal_type       80187 non-null  object
 3   breed             80187 non-null  object
 4   color             80187 non-null  object
 5   datetime          80187 non-null  object
 6   datetime2         80187 non-null  object
 7   found_location    80187 non-null  object
 8   intake_condition  80187 non-null  object
 9   intake_type       80187 non-null  object
 10  name              55603 non-null  object
 11  sex_upon_intake   80186 non-null  object
dtypes: object(12)
memory usage: 7.3+ MB


In [7]:
intakes_outcomes.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 79672 entries, 0 to 79671
Data columns (total 41 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   age_upon_outcome            79672 non-null  object 
 1   animal_id_outcome           79672 non-null  object 
 2   date_of_birth               79672 non-null  object 
 3   outcome_subtype             36348 non-null  object 
 4   outcome_type                79662 non-null  object 
 5   sex_upon_outcome            79671 non-null  object 
 6   age_upon_outcome_(days)     79672 non-null  int64  
 7   age_upon_outcome_(years)    79672 non-null  float64
 8   age_upon_outcome_age_group  79672 non-null  object 
 9   outcome_datetime            79672 non-null  object 
 10  outcome_month               79672 non-null  int64  
 11  outcome_year                79672 non-null  int64  
 12  outcome_monthyear           79672 non-null  object 
 13  outcome_weekday             796

In [8]:
outcomes.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 80681 entries, 0 to 80680
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   age_upon_outcome  80673 non-null  object
 1   animal_id         80681 non-null  object
 2   animal_type       80681 non-null  object
 3   breed             80681 non-null  object
 4   color             80681 non-null  object
 5   date_of_birth     80681 non-null  object
 6   datetime          80681 non-null  object
 7   monthyear         80681 non-null  object
 8   name              56116 non-null  object
 9   outcome_subtype   36893 non-null  object
 10  outcome_type      80667 non-null  object
 11  sex_upon_outcome  80679 non-null  object
dtypes: object(12)
memory usage: 7.4+ MB


## Question 1: Is there an area where more pets are found?

In [9]:
grouped_by_location_found= intakes.groupby('found_location')
grouped_by_location_found.size().sort_values(ascending=False)[:15]

found_location
Austin (TX)                              14443
Outside Jurisdiction                       948
Travis (TX)                                921
7201 Levander Loop in Austin (TX)          517
Del Valle (TX)                             411
Pflugerville (TX)                          372
Manor (TX)                                 280
4434 Frontier Trl in Austin (TX)           163
124 W Anderson Ln in Austin (TX)           153
Leander (TX)                               113
12034 Research Blvd in Austin (TX)          98
1156 W Cesar Chavez in Austin (TX)          98
12034 Research in Austin (TX)               96
1834 Ferguson in Austin (TX)                90
1156 W Cesar Chavez St in Austin (TX)       74
dtype: int64

The top 5 areas where pets can be found are:
1: Travis (TX) with 921 pets
2: 7201 Levander Loop in Austin (TX) with 517 pets
3: Del Valle (TX) with 411 pets
4: Pflugerville (TX) wth 372 pets
5: Manor (TX) with 280 pets

Austin (TX) and Outside Jurisdiction are too broad to be considered as actual areas. I came to the result by grouping the found locations and sorting them by the highest number in descending order. 

## Question 2: What is the average number of pets found in a month in the year 2015? Are there months where there is a higher number of animals found?

In [10]:
average_number_month = intakes_outcomes[intakes_outcomes['intake_year'] == 2015]
average_number_month = average_number_month.count() / 12
average_number_month["animal_id_outcome"]

1558.25

In [11]:
intakes['datetime'] = pd.to_datetime(intakes['datetime'])
intakes['what_month'] = intakes['datetime'].dt.month_name()
intakes['what_year'] = intakes['datetime'].dt.year

month_found = intakes[intakes['what_year'] == 2015].groupby('what_month')
top_months_found = month_found.size().sort_values(ascending=False)[:5]

month_found
top_months_found

what_month
June       2189
May        2094
October    1740
August     1718
July       1635
dtype: int64

The average of pets found in 2015 was roughly 1558. I got to that number by simply counting the intakes of the year 2015 and dividing them by 12. To evaluate the highest months regarding animals found I used the datetime method "dt" and set the year to 2015. After that I filtered the top months in which animals are adopted, which are as follows:June 2189, May 2094, October 1740 , August 1718 and July 1635.

## Question 3: What is the ratio of incoming pets vs. adopted pets?

In [12]:
adaptions = outcomes[outcomes['outcome_type'] == "Adoption"]
ratio = intakes['animal_id'].count() / adaptions.count()
ratio = 100 / (ratio["animal_id"])
ratio.round(2)

42.69

The ratio of incoming pets and adopted pets is 42.69%. This is fairly straight forward and one gets to that number by counting the incoming pets and dividing them by the amount of total adoptions.

## Question 4: What is the distribution of the types of animals in the shelter?

In [13]:
count_by_animal = intakes.groupby('animal_type').size().sort_values(ascending=False)
count_by_animal

animal_type
Dog          45743
Cat          29659
Other         4434
Bird           342
Livestock        9
dtype: int64

The distribution of animal types in the shelter is:
- Dogs 45743
- Cats 29659
- Other 4434
- Birds 342
- Livestock 9

I was able to find the result by grouping the animal type, showing the amount(which is size) of each animal type and sorting those by size as well.

## Question 5: What are the adoption rates for specific breeds?

In [14]:
top_breeds = outcomes[outcomes['animal_type'] == 'Dog'].groupby('breed').size().sort_values(ascending=False)[:5]
for label, value in top_breeds.items():
    dogs_adopted = len(outcomes[(outcomes['breed'] == label) & (outcomes['outcome_type'] == 'Adoption')])
    print(label, round((dogs_adopted/value)*100), "%")
    

Pit Bull Mix 37 %
Chihuahua Shorthair Mix 47 %
Labrador Retriever Mix 50 %
German Shepherd Mix 48 %
Australian Cattle Dog Mix 56 %


The adoption rates for the top 5 breeds were Pit Bull Mix 37%, Chihuahua Shorthair Mix 47 %, Labrador Retriever Mix 50 %, German Shepherd Mix 48% and Australian Cattle Dog Mix 56 %. I found the top 5 breeds by grouping the breed and filtering by the top 5 results. Then I used a for loop to check the top breeds and give me the outcome rate for those breeds compared to the adoption rate.

## Question 6: What are the adoption rates for different colorings?

In [15]:
top_colorings = outcomes.groupby('color').size().sort_values(ascending=False)[:5]

for label, value in top_colorings.items():
    adopted_colorings = len(outcomes[(outcomes['color'] == label) & (outcomes['outcome_type'] == 'Adoption')])
    print(label, round((adopted_colorings/value)*100), "%")


Black/White 45 %
Black 41 %
Brown Tabby 42 %
Brown 22 %
White 38 %


The adoption for the top 5 colorings are:
Black/White 45 %
Black 41 %
Brown Tabby 42 %
Brown 22 %
White 38 %

Similar to question 5, I used the groupy method and a for loop, to evaluate the adoption rate of the top five colorings.

## Question 7: About how many animals are spayed/neutered each month?

In [16]:
sex_types = intakes['sex_upon_intake'].unique()
sex_types

array(['Spayed Female', 'Intact Male', 'Intact Female', 'Neutered Male',
       'Unknown', nan], dtype=object)

In [72]:
total_intact = intakes[(intakes['sex_upon_intake'] == 'Intact Female') | (intakes['sex_upon_intake'] == 'Intact Male')].groupby('what_year').size()
total_intact_updated = total_intact.drop(labels = [2013, 2018])
average_per_month = (total_intact_updated.sum())/((len(total_intact_updated))*12)
print(average_per_month.round())

933.0


The average animals sprayed or neutered each month is 933. I used the unique method to evaluate the different sex types upon intake. Assuming that all intact males and females will be sprayed or neutered I grouped them by year found but excluded the years 2013 and 2018 since they are incomplete, so we have 4 total year to count from. The sum function counts all the incact animals in those 4 yours and divides them by the number of years multiplied by the numbers of months per year. That will lead to the result of 933 animals sprayed or neutered per month.